pythonforandroid/bootstrap.py

```diff
@@ -138,20 +141,22 @@ def name(self):
         return modname.split(".", 2)[-1]
 
     def prepare_build_dir(self):
-        '''Ensure that a build dir exists for the recipe. This same single
-        dir will be used for building all different archs.'''
+        """Ensure that a build dir exists for the recipe. This same single
+        dir will be used for building all different archs."""
         self.build_dir = self.get_build_dir()
-        self.common_dir = self.get_common_dir()
-        copy_files(join(self.bootstrap_dir, 'build'), self.build_dir)
-        copy_files(join(self.common_dir, 'build'), self.build_dir,
-                   override=False)
-        if self.ctx.symlink_java_src:
-            info('Symlinking java src instead of copying')
-            shprint(sh.rm, '-r', join(self.build_dir, 'src'))
-            shprint(sh.mkdir, join(self.build_dir, 'src'))
-            for dirn in listdir(join(self.bootstrap_dir, 'build', 'src')):
-                shprint(sh.ln, '-s', join(self.bootstrap_dir, 'build', 'src', dirn),
```

Comment:
This is the reason the `test_bootstrap_prepare_build_dir_with_java_src()` unit test is failing. It's because we no longer call `sh.ln` in this `prepare_build_dir()` function. So it's fine, we just need to remove the mock in the unit test for this one. So basically you need to edit the `test_bootstrap_prepare_build_dir_with_java_src()` test and remove the `sh.ln` patching as well as the assert we're doing with it and we should be all good